src/test/java/net/syscon/elite/service/CourtHearingsServiceTest.java

```diff
@@ -108,22 +128,130 @@ void setup() {
+    }
 
-    // TODO add test for non-matching court case
+    private void givenNoActiveCourtCase() {
+        when(offenderBooking.isActive()).thenReturn(true);
+        when(offenderBookingRepository.findById(OFFENDER_BOOKING_ID)).thenReturn(Optional.of(offenderBooking));
+        when(offenderBooking.getCourtCaseBy(COURT_HEARING.getCourtCaseId())).thenReturn(Optional.of(offenderCourtCase));
+        when(offenderCourtCase.isActive()).thenReturn(false);
+    }
+
+    @Test
+    void scheduleHearing_errors_when_prison_not_found() {
+        givenPrisonNotFound();
+
+        assertThatThrownBy(() -> courtHearingsService.scheduleHearing(OFFENDER_BOOKING_ID, COURT_HEARING))
+                .isInstanceOf(EntityNotFoundException.class)
+                .hasMessageContaining("Resource with id [PRISON] not found.");
+    }
+
+    private void givenPrisonNotFound() {
+        when(offenderBooking.isActive()).thenReturn(true);
+        when(offenderBookingRepository.findById(OFFENDER_BOOKING_ID)).thenReturn(Optional.of(offenderBooking));
+        when(offenderBooking.getCourtCaseBy(COURT_HEARING.getCourtCaseId())).thenReturn(Optional.of(offenderCourtCase));
+        when(offenderCourtCase.isActive()).thenReturn(true);
+        when(agencyLocationRepository.findById("PRISON")).thenReturn(Optional.empty());
+    }
+
+    @Test
+    void scheduleHearing_errors_when_court_not_found() {
+        givenCourtNotFound();
+
+        assertThatThrownBy(() -> courtHearingsService.scheduleHearing(OFFENDER_BOOKING_ID, COURT_HEARING))
+                .isInstanceOf(EntityNotFoundException.class)
+                .hasMessageContaining("Resource with id [COURT] not found.");
+    }
+
+    private void givenCourtNotFound() {
+        when(offenderBooking.isActive()).thenReturn(true);
+        when(offenderBooking.getLocation()).thenReturn(fromPrison);
+        when(offenderBookingRepository.findById(OFFENDER_BOOKING_ID)).thenReturn(Optional.of(offenderBooking));
+        when(offenderBooking.getCourtCaseBy(COURT_HEARING.getCourtCaseId())).thenReturn(Optional.of(offenderCourtCase));
+        when(offenderCourtCase.isActive()).thenReturn(true);
+        when(agencyLocationRepository.findById("PRISON")).thenReturn(Optional.of(fromPrison));
+        when(agencyLocationRepository.findById("COURT")).thenReturn(Optional.empty());
+    }
+
+    @Test
+    void scheduleHearing_errors_when_supplied_court_is_not_court() {
+        givenProvidedCourtLocationIsNotACourt();
+
+        assertThatThrownBy(() -> courtHearingsService.scheduleHearing(OFFENDER_BOOKING_ID, COURT_HEARING))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Supplied court location wih id COURT is not a valid court location.");
+    }
+
+    private void givenProvidedCourtLocationIsNotACourt() {
+        when(offenderBooking.isActive()).thenReturn(true);
+        when(offenderBooking.getLocation()).thenReturn(fromPrison);
```

Comment:
mocking out a load of stuff - isn't it easier to instantiate these objects using their builders and populate so they are in the correct state?

Reply:
Yes I know.  Gonna tidy this up. I realise it is ugly.

Reply:
Theres a nice pattern they use in community-api. 
Create a method that produces an exemplar  
Then in test call method, then `toBuilder` and configure object with bits salient for test:

e.g: https://github.com/ministryofjustice/community-api/blob/master/src/test/java/uk/gov/justice/digital/delius/service/StaffServiceTest.java#L147

Reply:
OK well I had already made changes.  I think will suffice for now (hint hint).